pkg/apis/extensions/types.go

```diff
@@ -905,6 +905,11 @@ type PodSecurityPolicySpec struct {
 	// may be used.
 	// +optional
 	Volumes []FSType
+	// AllowedFlexVolumes is a whitelist of allowed Flexvolumes.  Empty or nil indicates that all
+	// Flexvolumes may be used.  This parameter is effective only when the usage of the Flexvolumes
+	// is allowed in the "Volumes" field.
+	// +optional
+	AllowedFlexVolumes []AllowedFlexVolume
```

Comment:
nit: place after AllowedHostPaths

Reply:
There is the potential for the volume restrictions to get out of hand... not necessarily in this PR, but I think we need to restructure these (maybe in PSP `v1beta2`?)
xref: https://github.com/kubernetes/kubernetes/issues/29896